File: solutions/lua/space-age/2/space-age.lua
local SpaceAge = {}

local factors = {
  on_earth = 1,
  on_mercury = 0.2408467,
  on_venus = 0.61519726,
  on_mars = 1.8808158,
  on_jupiter = 11.862615,
  on_saturn = 29.447498,
  on_uranus = 84.016846,
  on_neptune = 164.79132,
}
function SpaceAge:new(seconds)
  self.__index = function(this, key)
    local on_earth = this.seconds / (60 * 60 * 24 * 365.25)
    return function() return tonumber(("%.2f"):format(on_earth / factors[key])) end
  end
  return setmetatable({ seconds = seconds }, self)
end

return SpaceAge

File: solutions/cpp/space-age/2/space_age.cpp
#include "space_age.h"

namespace space_age {
    constexpr int one_earth_year_in_seconds = 31557600;
    constexpr double one_mercury_year_in_earth_years = 0.24084670;
    constexpr double one_venus_year_in_earth_years = 0.61519726;
    constexpr double one_mars_year_in_earth_years = 1.88081580;
    constexpr double one_jupiter_year_in_earth_years = 11.86261500;
    constexpr double one_saturn_year_in_earth_years = 29.44749800;
    constexpr double one_uranus_year_in_earth_years = 84.01684600;
    constexpr double one_neptune_year_in_earth_years = 164.79132000;

    space_age::space_age(std::int64_t age_in_seconds) : age_in_seconds_(age_in_seconds) {}

    std::int64_t space_age::seconds() const { return age_in_seconds_; }

    double space_age::on_earth() const { return static_cast<double>(seconds()) / one_earth_year_in_seconds; }
    double space_age::on_mercury() const { return on_earth() / one_mercury_year_in_earth_years; }
    double space_age::on_venus() const { return on_earth() / one_venus_year_in_earth_years; }
    double space_age::on_mars() const { return on_earth() / one_mars_year_in_earth_years; }
    double space_age::on_jupiter() const { return on_earth() / one_jupiter_year_in_earth_years; }
    double space_age::on_saturn() const { return on_earth() / one_saturn_year_in_earth_years; }
    double space_age::on_uranus() const { return on_earth() / one_uranus_year_in_earth_years; }
    double space_age::on_neptune() const { return on_earth() / one_neptune_year_in_earth_years; }
}  // namespace space_age
